File: AlgoExpert/Easy/Transpose_Matrix/CPP/Transpose_Matrix.cpp
/*
Problem Name: Transpose Matrix
Difficulty: Easy
Platform: AlgoExpert

Problem Description:
Given a 2D array of integers (matrix), return the transpose of the matrix.

Time Complexity: O(n * m)
Space Complexity: O(n * m)
*/


#include <vector>
using namespace std;


vector<vector<int>> transposeMatrix(const vector<vector<int>>& matrix) {
    int rows = matrix.size();
    int cols = matrix[0].size();
    vector<vector<int>> transposedMatrix(cols, vector<int>(rows));

    for (int col = 0; col < cols; ++col) {
        for (int row = 0; row < rows; ++row) {
            transposedMatrix[col][row] = matrix[row][col];
        }
    }
    return transposedMatrix;
}

File: AlgoExpert/Easy/Transpose_Matrix/Python/Transpose_Matrix.py
"""
Problem Name: Transpose Matrix
Difficulty: Easy
Platform: AlgoExpert

Problem Description:
Given a 2D array of integers (matrix), return the transpose of the matrix.

Time Complexity: O(n * m)
Space Complexity: O(n * m)
"""

def transposeMatrix(matrix):
    transposedMatrix = []
    for col in range(len(matrix[0])):
        newRow = []
        for row in range(len(matrix)):
            newRow.append(matrix[row][col])
        transposedMatrix.append(newRow)
    return transposedMatrix

